astropy/utils/data.py

```diff
@@ -1371,5 +1504,92 @@ def get_cached_urls():
         warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
         return False
 
-    with shelve.open(urlmapfn) as url2hash:
-        return list(url2hash.keys())
+    try:
+        with shelve.open(urlmapfn, "r") as url2hash:
+            return list(url2hash.keys())
+    except dbm.error:
+        return []
+
+
+_cache_zip_index_name = "index.json"
+
+
+def export_download_cache(filename_or_obj, urls=None):
+    """Exports the cache contents as a ZIP file
+
+    Parameters
+    ----------
+    filename_or_obj : str or file-like
+        Where to put the created ZIP file. Must be something the zipfile
+        module can write to.
+    urls : iterable of str or None
+        The URLs to include in the exported cache. The default is all
+        URLs currently in the cache. If a URL is included in this list
+        but is not currently in the cache, it will be downloaded to the
+        cache first and then exported.
+
+    See Also
+    --------
+    import_download_cache : import the contents of such a ZIP file into the cache
+    """
+    if urls is None:
+        urls = get_cached_urls()
+    prefix = "cache"
+    with zipfile.ZipFile(filename_or_obj, 'w',
+                         compression=zipfile.ZIP_DEFLATED) as z:
+        index = {}
+        for u in urls:
+            # timeout=0 might prevent going to the Net if the url has
```

Comment:
The comment is unclear to someone coming to the code for the first time - is this a TODO?